lib/Dialect/FIRRTL/Transforms/CheckUninferredResets.cpp

```diff
@@ -0,0 +1,60 @@
+//===- CheckUninferredResets.cpp - Verify no uninferred resets ------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This pass checks to see if there are abstract type resets which have not
+// been inferred correctly.
+//
+//===----------------------------------------------------------------------===//
+
+#include "circt/Dialect/FIRRTL/FIRRTLOpInterfaces.h"
+#include "circt/Dialect/FIRRTL/FIRRTLUtils.h"
+#include "circt/Dialect/FIRRTL/Passes.h"
+#include "mlir/Pass/Pass.h"
+
+namespace circt {
+namespace firrtl {
+#define GEN_PASS_DEF_CHECKUNINFERREDRESETS
+#include "circt/Dialect/FIRRTL/Passes.h.inc"
+} // namespace firrtl
+} // namespace circt
+
+using namespace mlir;
+using namespace circt;
+using namespace firrtl;
+
+namespace {
+struct CheckUninferredResetsPass
+    : public circt::firrtl::impl::CheckUninferredResetsBase<
+          CheckUninferredResetsPass> {
+  void runOnOperation() override;
+};
+} // namespace
+
+void CheckUninferredResetsPass::runOnOperation() {
+  bool hasAbstractResetPorts = false;
+  for (FModuleLike module :
+       getOperation().getBodyBlock()->getOps<FModuleLike>()) {
+    for (PortInfo port : module.getPorts()) {
+      if (getBaseOfType<ResetType>(port.type)) {
+        auto diag = emitError(port.loc)
+                    << "a port \"" << port.getName()
+                    << "\" with abstract reset type was unable to be "
+                       "inferred by InferResets (is this a top-level port?)";
+        diag.attachNote(module->getLoc())
+            << "the module with this uninferred reset port was defined here";
+        hasAbstractResetPorts = true;
```

Comment:
You should be able to immediately fail here if you're emitting an error, although I do understand what you're trying to do here by marking all of the abstract resets as errors and not just the first one.